aten/src/ATen/native/PointwiseOps.cpp

```diff
@@ -9,39 +9,26 @@
 #include <ATen/NamedTensorUtils.h>
 
 namespace at {
-namespace native {
+namespace meta {
 
-Tensor addcmul(
-    const Tensor& self,
-    const Tensor& tensor1,
-    const Tensor& tensor2,
-    const Scalar& value) {
-  Tensor result = at::empty({0}, self.options());
-  return at::addcmul_out(result, self, tensor1, tensor2, value);
+TORCH_META_FUNC(addcmul)
+(const Tensor& self,
+ const Tensor& tensor1,
+ const Tensor& tensor2,
+ const Scalar& value) {
+  build_ternary_op(maybe_get_output(), self, tensor1, tensor2);
 }
 
-Tensor& addcmul_(
-    Tensor& self,
-    const Tensor& tensor1,
-    const Tensor& tensor2,
-    const Scalar& value) {
-  return at::addcmul_out(self, self, tensor1, tensor2, value);
-}
+} // namespace meta
+namespace native {
 
-Tensor& addcmul_out(const Tensor& self,
-    const Tensor& tensor1,
-    const Tensor& tensor2,
-    const Scalar& value,
-    Tensor& result) {
-  checkBackend("addcmul_cpu", result, self.options().backend());
```

Comment:
We're losing this, but it looks like it's redundant since TensorIterator defaults to checking that all input/output tensors are on the same device (which is equivalent to the backend check). We also want to deprecate `backend` at some point in the future anyway :)

Reply:
Doesn't `set_output` checks that already?

Reply:
yep!